source/elements/oneMKL/source/architecture/other_architecture.inc.rst

```diff
@@ -11,10 +11,41 @@ This section covers all other features in the design of oneMKL architecture.
 
 .. _onemkl_spec_current_version:
 
-Current Version of this oneMKL Specification
+Specification Version and Compliance
 +++++++++++++++++++++++++++++++++++++++++++++
 
-This is the oneMKL specification which is part of the oneAPI specification version 1.0.0.
+Each oneMKL domain must define a preprocessor macro to represent the version of the specification that the implementation is compliant with.
```

Comment:
Do all headers provide this info? E.g.,

- Will `#include "oneapi/mkl.hpp"` provide `ONEMKL_*_SPEC_VERSION` for all domains?
  - If a domain is not supported/installed, will it be undefined of `0`?
- Will `#include "oneapi/mkl/rng.hpp"`  provide `ONEMKL_RNG_SPEC_VERSION`?
- Will `#include "oneapi/mkl/rng.hpp"` provide `ONEMKL_BLAS_SPEC_VERSION`?
- Will `oneapi/mkl/rng/device.hpp` provide `ONEMKL_RNG_SPEC_VERSION`?

Reply:
@al42and good question. The specification(and this RFC) does not define any implementation-specific header files for oneMKL, so the location of the macro can be decided by the implementation itself.

> If a domain is not supported/installed, will it be undefined or 0?

For an undefined domain implementation: https://oneapi-src.github.io/oneMKL/create_new_backend.html#create-wrappers shows how the wrappers in oneMKL Interfaces work and how the implementation can handle errors for unimplemented backends. Note that this is different than the oneAPI specification that we have here. More clarification on that: https://github.com/oneapi-src/oneMKL/blob/develop/README.md#faqs

I added a rule for the macro numbering for the non-compliance of an implementation. Thanks!

Reply:
> For an undefined domain implementation, this, oneapi-src.github.io/oneMKL/create_new_backend.html#create-wrappers shows how the wrappers in oneMKL Interfaces work and how the implementation can handle errors for unimplemented backends.

I was referring not to the case of unimplemented _backends_, but to the case of a _domain_ missing entirely: "It is permitted for a library implementation of the oneMKL specification to implement either all, one or more than one of the domains in oneMKL."

But now that I read it, the wording "Each oneMKL domain must define a preprocessor macro [...]" is quite clear that if domain is not supported, the macro is not defined.

Reply:
@al42and yes, you are right.